### PR TITLE
finished task 6

### PR DESCRIPTION
**Pull request by a-stenhouse**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -49,4 +49,40 @@ describe.only("get /api/reviews", () => {
...
             })
     })
+})
+
+describe("get /api/reviews/:review_id/comments", () => {
+    test("responds with JSON containig an array of comments assosciated with the given review_id", () => {
+        return request(app)
+            .get("/api/reviews/3/comments")
+            .expect(200)
+            .then((response) => {
```
Comment (Augs0) [inline __tests__/app.test.js:96]: We should always check the length of the array before iterating over it

File: __tests__/app.test.js [hunk 2]
```diff
@@ -49,4 +49,40 @@ describe.only("get /api/reviews", () => {
...
+    });
+    test("status:400, reponds with an error message when passed a bad review_id", () => {
+        return request(app)
+            .get("/api/reviews/notAReviewID/comments")
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe("Bad request")
+            })
+    })
 })
```
Comment (Augs0) [inline __tests__/app.test.js:227]: We're missing one test here for a valid review with no comments that should return an empty array

File: controller.js [hunk 3]
```diff
@@ -12,3 +12,9 @@ exports.getReviews = (request, response, next) => {
         .then((reviews) => response.status(200).send({ reviews }))
         .catch(next);
 };
+
+exports.getReviewComments = (request, response, next) => {
+    return fetchReviewComments(request.params.review_id)
+        .then((comments) => response.status(200).send({ comments }))
```
Comment (Augs0) [inline controller.js:26]: HINT: to handle the empty comments you will need to do something different here ;)

**Review by Augs0 — CHANGES_REQUESTED** [review]
Almost there :) just missing one test that checks for no comments on a review. Please see the feedback

File: controller.js (hunk 3 above)
Comment (a-stenhouse) [reply controller.js:26]: Hi, if it should still send an object containing an empty array if there are no comments, do I need to change this? Only reason I can see is if I need to send a different status code. If that's the case, is it a 204 (No content)?

**Review by pippaamy — COMMENTED** [review]
Looks good, still need to check array length before the beforeEach in test!
